enclave_wrangler/new_enclave_api.py

```diff
@@ -0,0 +1,540 @@
+  TO-DO: find out whats required by doing a validate post
+    - There should be API examples in response.json i got for this
+    - Probably don't need the random ID we were using for new concept set / version
+    - 1 & 2 are the most important.
+    - Curl first and command at top of function for easy testing
+    - For unit test, use validate url?
+  0. Test case file. Might want to write a unit test for this.
+    - test_enclave.py. 3 palantir files; just a single concept set.
+  1. Create a new concept set
+    1.1. Create a new concept set (see: createNewConceptSetDescription; "apiName": "create-new-concept-set",)
+    1.2. Create new draft (see: createNewDraft; "apiName": "create-new-draft-omop-concept-set-version",)
+      - This will be empty at first; the first draft.
+    1.3. Add concepts to the draft.
+    1.4. Approve: Turn draft into a version.
+    Once first is working, should be able to figure out the second.
+  2. Add/remove concepts from existing concept set
+    - (Siggie doesn't know the steps)
+  3. Change metadata on existing concept set
+    - (Siggie doesn't know the steps)
+
+todo's (later/minor/contingent)
+  - to-do: contingent: @amin/enclave: Concept set ownership. Right now, personal access token is required. Service token
+  (i.e. for bulk import) not allowed. This may create an issue later where, when a user is using TermHub and they upload
+  a concept set, we will have to (a) use one of our personal access tokens, in which case it will not show in the
+  enclave who really uploaded the concept set; it'll say one of us, or (b) we have the user pass their own personal
+  access token. Amin (2022/10/19) said: "decision: onBehalfOf". I suppose this means that we'll add this optional param
+  instead of an optional 'owner' param.
+  - to-do: would we indeed represent OntologyObject types as Dict?
+
+Notes
+- Can't delete a concept-set-container. This is to prevent an accident where we delete something that someone else
+depends on. This may change in the future; Amin could allow us to delete a specific concept set with a known name.
+However, archive-concept-set is good enough for now.
+    {
+      "apiName": "",
+      "description": "Sets Concept Set 'archived' property to true so that it no longer appears in browser",
+      "rid": "ri.actions.main.action-type.cbc3643b-cca4-4772-ae2d-ae7036a6798b",
+      "parameters": {
+        "concept-set": {
+          "description": "",
+          "baseType": "OntologyObject"
+        }
+      }
+    },
+
+TODO: Remove this temporary list of API endpoints when done / if advisable
+"""
+import sys
+from typing import Dict, List, Union
+
+import requests
+
+from enclave_wrangler.config import config
+
+
+JSON_TYPE = Union[List, Dict]
+
+
+def upload_concept(
+    include_descendants: bool, concept_set_version_item: str, is_excluded: bool, include_mapped: bool, validate=False
+) -> JSON_TYPE:
+    """Create new concepets within concept set
```

Comment:
I've added a lot of documentation, i.e. from the one API call that shows API definitions, as well as validation info, here and in comments next to each param.